modules/lang-painless/src/main/java/org/elasticsearch/painless/phase/DefaultSemanticAnalysisPhase.java

```diff
@@ -0,0 +1,2728 @@
+        AExpression userConditionNode = userIfNode.getConditionNode();
+        semanticScope.setCondition(userConditionNode, Read.class);
+        semanticScope.putDecoration(userConditionNode, new TargetType(boolean.class));
+        checkedVisit(userConditionNode, semanticScope);
+        decorateWithCast(userConditionNode, semanticScope);
+
+        SBlock userIfBlockNode = userIfNode.getIfBlockNode();
+
+        if (userConditionNode instanceof EBooleanConstant || userIfBlockNode == null) {
+            throw userIfNode.createError(new IllegalArgumentException("extraneous if block"));
+        }
+
+        semanticScope.replicateCondition(userIfNode, userIfBlockNode, LastSource.class);
+        semanticScope.replicateCondition(userIfNode, userIfBlockNode, InLoop.class);
+        semanticScope.replicateCondition(userIfNode, userIfBlockNode, LastLoop.class);
+        visit(userIfBlockNode, semanticScope.newLocalScope());
+        semanticScope.replicateCondition(userIfBlockNode, userIfNode, AnyContinue.class);
+        semanticScope.replicateCondition(userIfBlockNode, userIfNode, AnyBreak.class);
+    }
+
+    @Override
+    public void visitIfElse(SIfElse userIfElseNode, SemanticScope semanticScope) {
+        AExpression userConditionNode = userIfElseNode.getConditionNode();
+        semanticScope.setCondition(userConditionNode, Read.class);
+        semanticScope.putDecoration(userConditionNode, new TargetType(boolean.class));
+        checkedVisit(userConditionNode, semanticScope);
+        decorateWithCast(userConditionNode, semanticScope);
+
+        SBlock userIfBlockNode = userIfElseNode.getIfBlockNode();
+
+        if (userConditionNode instanceof EBooleanConstant || userIfBlockNode == null) {
+            throw userIfElseNode.createError(new IllegalArgumentException("extraneous if block"));
+        }
+
+        semanticScope.replicateCondition(userIfElseNode, userIfBlockNode, LastSource.class);
+        semanticScope.replicateCondition(userIfElseNode, userIfBlockNode, InLoop.class);
+        semanticScope.replicateCondition(userIfElseNode, userIfBlockNode, LastLoop.class);
+        visit(userIfBlockNode, semanticScope.newLocalScope());
+
+        SBlock userElseBlockNode = userIfElseNode.getElseBlockNode();
+
+        if (userElseBlockNode == null) {
+            throw userIfElseNode.createError(new IllegalArgumentException("extraneous else block."));
+        }
+
+        semanticScope.replicateCondition(userIfElseNode, userElseBlockNode, LastSource.class);
+        semanticScope.replicateCondition(userIfElseNode, userElseBlockNode, InLoop.class);
+        semanticScope.replicateCondition(userIfElseNode, userElseBlockNode, LastLoop.class);
+        visit(userElseBlockNode, semanticScope.newLocalScope());
+
+        if (semanticScope.getCondition(userIfBlockNode, MethodEscape.class) &&
+                semanticScope.getCondition(userElseBlockNode, MethodEscape.class)) {
```

Comment:
Line these conditions up.

Reply:
All five of 'em.

Reply:
I lined them up. Note that standard style dictates a line continuation is intended twice normally.